phygnn/base.py

```diff
@@ -497,19 +498,37 @@ def load(cls, fpath):
 
         with open(fpath, 'rb') as f:
             model_params = pickle.load(f)
+            model_params = cls._history_to_df(model_params)
 
         if 'version_record' in model_params:
             version_record = model_params.pop('version_record')
             logger.info('Loading model from disk that was created with the '
                         'following package versions: \n{}'
                         .format(pprint.pformat(version_record, indent=4)))
 
+        sig = signature(cls)
+        model_params = {k: v for k, v in model_params.items()
+                        if k in sig.parameters}
         model = cls(**model_params)
         logger.info('Successfully initialized model from file: {}'
                     .format(fpath))
 
         return model
 
+    @classmethod
+    def _history_to_dict(cls, model_params):
+        """Make sure history is a dictionary prior to saving"""
+        if model_params.get('history', None) is not None:
+            model_params['history'] = dict(model_params['history'])

```

Comment:
you should check this operation. I don't think it does what you want it to. Recommend using `DataFrame.to_dict()` instead.

Reply:
hmm. this is how I manually converted the history in the exisitng pickle files but I'll change it.

Reply:
it might work but it accomplish our goal of not saving pandas objects:

<img width="319" alt="image" src="https://user-images.githubusercontent.com/12700356/231852249-0305852a-332f-4501-b952-8a694b92aebf.png">

Reply:
Yeah just checked that. Good to know